crates/red_knot_module_resolver/src/testing.rs

```diff
@@ -120,6 +121,16 @@ impl<T> TestCaseBuilder<T> {
         self
     }
 
+    /// Add other files external to any of the pre-configured search paths.
+    ///
+    /// (For example, an editable installation in `site-packages`
+    /// might point to a directory outside of any of the typing spec's
+    /// listed module resolution sources)
+    pub(crate) fn with_other_files(mut self, files: &[FileSpec]) -> Self {
```

Comment:
They can be any files, yes. Editable installs can point to directories that are outside any of the "static" module-resolution root paths, so we need a way of modelling that in tests. The files are written to wherever the path points: `.with_other_file([("/foo/bar/src/baz.py", "")])` adds an empty file at location `/foo/bar/src/baz.py`.

Agree it's not a great name... any ideas for how to improve it? :P

Reply:
Fair enough

Reply:
Done in https://github.com/astral-sh/ruff/pull/12307/commits/1adf37d575a6d3571389a8fae3503f41060b5da7, though in my opinion it makes the test code somewhat uglier than it would be otherwise :(